config/confighttp/confighttp.go

```diff
@@ -184,11 +182,6 @@ func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Ext
 	}, nil
 }
 
-// ToClientWithHost creates an HTTP client.
```

Comment:
IIUC the goal is that we don't remove or modify anything without it being deprecated before, so we need to keep `ToClientWithHost` for one version while being deprecated

Reply:
Done